lib/multi.c

```diff
@@ -1392,9 +1392,15 @@ static CURLMcode multi_runsingle(struct Curl_multi *multi,
       }
 
       if(!result) {
-        if(async)
+        if(async) {
           /* We're now waiting for an asynchronous name lookup */
           multistate(data, CURLM_STATE_WAITRESOLVE);
+#ifdef HAVE_SOCKETPAIR
+          /* return CURLM_CALL_MULTI_PERFORM so that client can
+             query read socket fd for polling */
+          rc = CURLM_CALL_MULTI_PERFORM;
```

Comment:
The comment is incorrect, as this return value won't be handled by the "client", it will just cause an internal loop.

Reply:
I had added this to let application call curl_multi_perform() based on return status to see if DNS is resolved but as you said this is not required. I have tested with TCP capture and confirmed connection request is sent with in 10 msec after DNS resolution. Hence, removed this change.

Reply:
libcurl never returns that anyway, it is only used for an internal loop

Reply:
You are right, libcurl doesn't return this status.
It seems earlier version of libcurl was returning this status as my application is still handling CURLM_CALL_MULTI_PERFORM status and needs to be correct.
Thanks for the clarification !